### DISCUSSION TITLE
Challenge not detected when showing in Selenium browser

### DISCUSSION THREAD

**Discussion by bk1n** [FlareSolverr/FlareSolverr#755] [Q&A]
Hi all,

I am trying to parse https://www.hltv.org/stats/players, player by player. I am running into constant Cloudflare issues, as they have just upgraded their DDoS protection. 

I have followed this guide here to try to set up FlareSolverr: https://scrapeops.io/python-web-scraping-playbook/python-flaresolverr/#install--run-flaresolverr

I have created a function in Python that looks like this, sending a post request to the FlareSolverr server once it's been set up
```Python
    def _pass_cloudflare_challenge(self, url):
        '''Requires FlareSolverr to be running!'''
        post_body = {
        "cmd": "request.get",
        "url": url,
        "maxTimeout": 60000
        }

        response = requests.post('http://localhost:8191/v1', headers={'Content-Type': 'application/json'}, json=post_body)

        if response.status_code == 200:
            json_response = response.json()
            print(f'-- FlareSolverr: {json_response["message"]}')
            if json_response.get('status') == 'ok':
                ## Get Cookies & Clean
                cookies = json_response['solution']['cookies']
                clean_cookies_dict = {cookie['name']: cookie['value'] for cookie in cookies}
                ## Get User-Agent
                user_agent = json_response['solution']['userAgent']
                ## Make normal request
                headers={"User-Agent": user_agent}
                return clean_cookies_dict, headers
```

Then, I use these cookies and header to run my own instance of Selenium
```Python
    def _get_selenium_driver(self, clean_cookies, headers):
        options = webdriver.ChromeOptions() 
        options.add_experimental_option('excludeSwitches', ['enable-logging'])
        options.add_argument('--disable-gpu')
        options.add_argument(f'user-agent={headers["User-Agent"]}')
        self.driver = webdriver.Chrome(options=options)
        self.driver.get(PLAYERS_URL)
        WebDriverWait(self.driver, 10).until(EC.element_to_be_clickable((By.ID, 'CybotCookiebotDialogBodyLevelButtonLevelOptinAllowAll'))).click()
        for name, value in clean_cookies.items():
            self.driver.add_cookie({'name': name, 'value': value})
```

I attempt to do the parsing, and will try to solve a new cloudfare challenge if the code throws an exception, and then relaunch a new Selenium instance
```Python
    def get_playerInfo(self):
        retry_limit = 0
        while True and retry_limit < 5:
            try:
                self.driver.get(url) 
               ### parsing ...
            except:
                retry_limit += 1
                cookies, header = self._pass_cloudflare_challenge(PLAYERS_URL)
                self._get_selenium_driver(cookies, header)
                continue
            break
```

This is not working, however. FlareSolverr is saying that there is no challenge detected, even when I can see a challenge in the browser. The challenge is a verify box that I can't solve (even when clicked with a mouse).

![Capture2](https://user-images.githubusercontent.com/75484578/229879661-21f85577-c62a-4916-a4b8-b0f2acb7cb21.PNG)
![Capture](https://user-images.githubusercontent.com/75484578/229879666-f4f262be-845c-42bf-a5b7-eba519534043.PNG)

The exact experience I get is: 

1. Selenium launches the driver
2. Navigates to the desired url (note, I can see the contents of the URL here, behind an Accept Cookies pop-up)
3. Accept cookies pop-up disappears
4. Website immediately goes to cloudflare verification
5. Python tries _pass_cloudflare_challenge() method; no challenge detected.
6. New browser launches and process repeats.

Can anyone help me? I may well be doing something wrong here, so apologies if that's the case.

**Comment by ngosang** [comment]
Do not use Selenium, it's detected by CloudFlare. You can use any script language like Python, JS...
